### PR TITLE
Convert remaining tests in test49 into new style tests

### PR DESCRIPTION
**Pull request by yegappan**
Remove test49 and the empty Makefile variables.

### CODE REVIEW

**Comment by k-takata** [comment]
Wow, amazing! All old tests are converted. (Except for the test for VMS?)

I think that makefiles for Windows can be simplified more.
https://gist.github.com/k-takata/99f21f641eae843df866eae729b2e2f5
`SCRIPTS_FIRST` is not used anymore.

**Comment by k-takata** [comment]
Maybe, `:help testing` needs to be updated.

**Comment by vim-ml** [comment]
Hi,

On Mon, Aug 10, 2020 at 6:59 PM K.Takata <vim-dev-github@256bit.org> wrote:

> Wow, amazing! All old tests are converted. (Except for the test for VMS?)
>

Yes. I don't have access to a VMS system to test the changes.
I can convert test77a to a new style test. But someone else needs
to test it on a VMS system.


> I think that makefiles for Windows can be simplified more.
> https://gist.github.com/k-takata/99f21f641eae843df866eae729b2e2f5
> SCRIPTS_FIRST is not used anymore.
>

Yes. I will remove SCRIPTS_FIRST also.

Regards,
Yegappan

File: src/testdir/Make_dos.mak [hunk 1]
```diff
@@ -72,12 +31,6 @@ report:
 		else ( echo ALL DONE )
 
 clean:
-	-del *.out
-	-del *.failed
-	-del *.res
-	-if exist $(DOSTMP) rd /s /q $(DOSTMP)
-	-if exist test.in del test.in
-	-if exist test.ok del test.ok
```
Comment (k-takata) [inline src/testdir/Make_dos.mak]: I'm thinking whether we should remove these lines from the `clean` target right now.
These files can be left behind when someone executed the tests in an old environment. Then when they update the source code, `make clean` won't clean the files.

**Comment by chrisbra** [comment]
Good job on getting rid of all the old tests! Amazing work, especially with test49.

File: src/testdir/Make_dos.mak [hunk 2]
```diff
@@ -31,6 +31,12 @@ report:
 		else ( echo ALL DONE )
 
 clean:
+	-del *.out
+	-del *.failed
+	-del *.res
+	-if exist $(DOSTMP) rd /s /q $(DOSTMP)
```
Comment (k-takata) [inline src/testdir/Make_dos.mak]:
```suggestion
	-if exist dostmp rd /s /q dostmp
```
`DOSTMP` is not defined now.

**Comment by brammool** [comment]
I'm surprised CI passes - the small.vim script is no longer generated, making all the other tests fail.
Apparently they don't make "make test".

**Comment by brammool** [comment]
Oh, the tests fail, but doesn't make the CI job fail.  I'll fix it.
We may also want to add some other minimal tests for the tiny version.  That can't be much thus we could put it in test1.in/test1.ok

**Comment by k-takata** [comment]
How about this?

```diff
--- a/src/testdir/runtest.vim
+++ b/src/testdir/runtest.vim
@@ -46,7 +46,9 @@
 
 
 " Without the +eval feature we can't run these tests, bail out.
-so small.vim
+silent! while 0
+  qa!
+silent! endwhile
 
 " In the GUI we can always change the screen size.
 if has('gui_running')
```

No need to run test1 for creating small.vim before running new style tests.

**Comment by brammool** [comment]
I would like to give the tiny and small builds at least some basic testing.  Just in case we manage to do something stupid, like a misplaced #ifdef causes them to exit, or .vimrc is not sourced, stuff like that.
E.g. create some text and write it in the output file.
Otherwise test1.in can be cleaned up, I don't think tiny.vim and mbyte.vim are used.

**Comment by k-takata** [comment]
I understand that. But, I don't think we need to run test1 before running new style tests.
For tiny builds, maybe creating a new make target is better? e.g. `make testtiny`

**Comment by brammool** [comment]
Yeah, it's cleaner if it can run separately.  Feel free to make a pull request.

**Comment by vim-ml** [comment]
On Mon, Aug 10, 2020 at 11:22 PM Christian Brabandt <
vim-dev-github@256bit.org> wrote:

> Good job on getting rid of all the old tests! Amazing work, especially
> with test49.
>
>
>
Thanks Christian.

Regards,
Yegappan

**Pull request closed without merging**